core/primitives/block_header.hpp

```diff
@@ -62,8 +63,11 @@ namespace kagome::primitives {
   template <class Stream,
             typename = std::enable_if_t<Stream::is_decoder_stream>>
   Stream &operator>>(Stream &s, BlockHeader &bh) {
-    return s >> bh.parent_hash >> bh.number >> bh.state_root
-           >> bh.extrinsics_root >> bh.digests;
+    boost::multiprecision::cpp_int number_compact;
```

Comment:
why don't we have BlockNumber as CompactInteger? just wondering. probably there is a reason.